### PR TITLE
fixed 449 and added a test

### PR DESCRIPTION
**Pull request by dyladan**
also, removes the GET parameters from the search phrase by splitting on the first '?'

### CODE REVIEW

File: bookie/views/api.py [hunk 1]
```diff
@@ -504,10 +504,10 @@ def search_results(request):
 
     """
     mdict = request.matchdict
-    rdict = request.GET
+    rdict = request.params
 
     if 'terms' in mdict:
-        phrase = " ".join(mdict['terms'])
+        phrase = " ".join(mdict['terms']).split("?")[0]
```
Comment (mitechie) [inline bookie/views/api.py]: this seems broken or  a bug. Pyramid should not include the query string in the matchdict. I'll have to test this out and see about working upstream. We upgraded to pyramid 1.5 recently and I wonder if there's a known bug or update to webob or pyramid we should be getting.

File: bookie/views/api.py [hunk 2]
```diff
@@ -504,10 +504,10 @@ def search_results(request):
 
     """
     mdict = request.matchdict
-    rdict = request.GET
+    rdict = request.params
```
Comment (mitechie) [inline bookie/views/api.py]: why the change here?

Comment (dyladan) [inline bookie/views/api.py]: did this as part of trying to pinpoint the issue. I'm pushing a commit now that switches it back

**Comment by dyladan** [comment]
amended the commit to switch back to using GET

File: bookie/views/api.py (hunk 1 above)
Comment (dyladan) [inline bookie/views/api.py]: not sure. i just know the GET parameters were being included in the terms and this is a working fix for that. If there is a better or more robust solution for this i'd be more than willing to do that instead

**Comment by dyladan** [comment]
see #449 closing

**Comment by dyladan** [comment]
this apparently does still fix the issue and the tests still pass. reopening

**Comment by mitechie** [comment]
Cool, can you rebase this back down to one commit and I'll test/qa this over the weekend. Thanks for looking into it.

**Comment by dyladan** [comment]
i was wrong. closing. The issue is still open, i'm just not sure what causes it.
